lib/model/message_list.dart

```diff
@@ -381,6 +381,7 @@ class MessageListView with ChangeNotifier, _MessageSequence {
     for (final message in result.messages) {
       if (_messageVisible(message)) {
         _addMessage(message);
+        store.recentSenders.handleMessage(message);
       }
```

Comment:
Reply to your [608 comment](https://github.com/zulip/zulip-flutter/pull/608#discussion_r1600478354):
> Good question.
>
> What does the dartdoc on that _messageVisible method say?
>
> Based on that, what would be the consequence of having that condition control whether to make this call? And what would be the consequence of ignoring that condition when making this call?

I concluded that the call should be inside the if condition. We track the messages that are shown in the feed, and if they're not shown, then there's no need to show the users who sent these messages, in the autocomplete results. Thanks for the clue!!

Reply:
If we look at web impl, [`process_stream_message`](https://github.com/zulip/zulip/blob/62dfd93a83b8604c53fcd7c6b00bfab90fa55d99/web/src/message_helper.js#L54) in [`process_new_message`](https://github.com/zulip/zulip/blob/62dfd93a83b8604c53fcd7c6b00bfab90fa55d99/web/src/message_helper.js#L11-L84) doesn't seem to care about topic visibility policy. Not sure if that's intended or a bug.

Reply:
Yeah, good comparison.

Here, `_messageVisible` says whether the message should appear in this particular message list:
```dart
  /// Whether [message] should actually appear in this message list,
  /// given that it does belong to the narrow.
  ///
  /// This depends in particular on whether the message is muted in
  /// one way or another.
  ///
  /// See also [_allMessagesVisible].
  bool _messageVisible(Message message) {
```

But if the message doesn't appear in this message list, it might still appear in a different one later. A message hidden from the combined feed because the stream is muted will appear if you navigate to that stream; a message hidden because the stream and/or topic is muted will appear if you navigate to the topic.

The recent-senders data structure isn't specific to a particular message list; it's shared across them all (within an account). So let's have the data it gets not be controlled by the message list's `_messageVisible`.